sktime/forecasting/reservoir.py

```diff
@@ -0,0 +1,225 @@
+        "requires-fh-in-fit": False,
+        "handles-missing-data": True,
+        "scitype:y": "univariate",
+        "ignores-exogeneous-X": True,
+    }
+
+    def __init__(self, regressor=None):
+        self.regressor = regressor
+        self.X_ = None
+        self.y_ = None
+        self.delay_ = 1
+        self.d_linear_ = None
+        self.d_nonlinear_ = None
+        self.fitted_y = False
+        super(NextRC, self).__init__()
+
+    def _feature_vector(self, y, k=None):
+        if k is None:
+            k = self.delay_
+
+        (n_samples, n_features) = y.shape
+
+        # size of the linear part of the feature vector
+        self.d_linear_ = d_linear = k * n_features
+        # size of the non-linear part of feature vector
+        self.d_nonlinear_ = d_nonlinear = int(
+            d_linear * (d_linear + 1) * (d_linear + 2) / 6
+        )
+        # total size
+        d = d_linear + d_nonlinear
+
+        # create an array to hold the linear part of the feature vector
+        features = np.zeros((d_linear, n_samples))
+
+        # fill in the linear part of the complete feature vector
+        for delay in range(k):
+            for j in range(delay, n_samples):
+                features[n_features * delay : n_features * (delay + 1), j] = y[
+                    j - delay, :
+                ]
+
+        # create full feature vector
+        O_features = np.zeros((d, n_samples))
+
+        # copy over the linear part
+        O_features[0:d_linear, :] = features
+
+        # fill in the non-linear part
+        cnt = 0
+        for row in range(d_linear):
+            for column in range(row, d_linear):
+                for span in range(column, d_linear):
+                    O_features[d_linear + cnt] = (
+                        O_features[row, :] * O_features[column, :] * O_features[span, :]
+                    )
+                    cnt += 1
+
+        return O_features
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y : pd.Series
+            Target time series with which to fit the forecaster.
+        X : pd.DataFrame, default=None
+            Exogenous variables are ignored
+        fh : int, list or np.array, optional (default=None)
+            The forecasters horizon with the steps ahead to to predict.
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        self.X_ = X
+        self.y_ = y
+        self.delay_ = 2
+
+        if self.regressor is None:
+            self.regressor_ = sklearn.linear_model.Ridge(
+                alpha=1.0e-3, fit_intercept=False
+            )
+        else:
+            self.regressor_ = clone(self.regressor)
+
+        n_features = y.shape[1]
+
+        # use relative fh as time index to predict
+        fh = self.fh.to_absolute_index(self.cutoff)
+
+        # print(y[: fh[-1] + 1, :])
+
+        X_features = self._feature_vector(y[: fh[-1] + 1, :])
```

Comment:
@frthjf could you help me know what fh[0] and fh[1] correspond to as per your code? In sktime fh[1] hit an error due to fh being `absolute` here. both fh[0] and fh[-1] gives 20 after changing `fh.to_absolute_index`

Reply:
If I remember correctly `fh[0]` is the start and `fh[1]` is the end of the interval used for prediction (index relative to the data array). So if you call `estimator.fit([1, 2, 3, 4, 5], fh=[2,3])` you are operating on `[3,4]`.

Reply:
I see. In `sktime`, a relative index of `0` is actually the `cutoff`, i.e., the last element of the array. `1` would correspond to an offset of one, i.e., if `y` seen in `fit` has length 20, with indices 0-19, then `1` would correspond to forecasting 20. `0` would correspond to an in-sample forecast of the index 19.